sympy/geometry/line.py

```diff
@@ -1070,6 +1071,48 @@ def contains(self, other):
             return Point.is_collinear(self.p1, self.p2, other.p1, other.p2)
         return False
 
+    def parameter_value(self, other):
+        """
+        Finds the parameter ``t`` for which
+        ``other == t * self.p1 + (1 - t) self.p2`` is True.
+
+        Raises
+        ======
+
+        ValueError
+            When ``other`` is not contained in the line.
+
+        NotImplementedError
+            When ``other`` is not a ``Point``.
+
+        Notes
+        =====
+
+        Do not confuse this function with function ``parameter`` in
+        sympy/geometry/curve.py .
+
+        Examples
+        ========
+
+        >>> from sympy import Point, Line
+        >>> p1, p2, p3 = Point(0, 1), Point(2, 3), Point(4, 5)
+        >>> l = Line(p1, p2)
+        >>> l.parameter_value(p3)
+        2
+        """
+        if isinstance(other, Point):
+            if Point.is_collinear(self.p1, self.p2, other):
+                t = Dummy("t")
+                x, y = self.arbitrary_point(t).args
+                if self.p1.x != self.p2.x:
+                    ti = list(solveset(x - other.x, t))[0]
```

Comment:
The solution `t` of the equation `x - other.x == 0`, or `self.p1.x + (self.p2.x - self.p1.x)*t == other.x` in expanded form, is easy to write down explicitly. There should be no need to call `solveset`.